assets/src/components/amp-support/notices.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	AMPNotice,
+	NOTICE_SIZE_SMALL,
+	NOTICE_TYPE_INFO,
+	NOTICE_TYPE_SUCCESS,
+	NOTICE_TYPE_WARNING,
+} from '../amp-notice';
+
+/**
+ * Render notices on site support page.
+ *
+ * @param {Object} props Component props.
+ * @return {JSX.Element|null} HTML markup for notice data.
+ */
+export function Notices( props ) {
+	const { data, args, ampValidatedPostCount } = props;
+
+	const isSpecificUrlRequest = ( 0 < args?.urls?.length );
+	const hasErrors = ( 0 < data?.errors?.length || 0 !== data?.urls?.length );
```

Comment:
1. Since `data.errors` is marked as `isRequired`, do we need the optional chaining here?
2. We could use the already determined `isSpecificUrlRequest` value here to make the code more readable.
3. There's no need for parentheses:

```suggestion
	const hasErrors = 0 < data.errors?.length || isSpecificUrlRequest;
```

Lastly, are we always sure that when the `args.urls` is non-empty then there are errors?

Reply:
> 1. Since `data.errors` is marked as `isRequired`, do we need the optional chaining here?
No, We can remove chaining.

> 2.  We could use the already determined `isSpecificUrlRequest` value here to make the code more readable.
There is a difference in condition here. For `isSpecificUrlRequest` we are check URLs length of `args` and for `hasErrors` we need to check URLs the length of data.

> Lastly, are we always sure that when the `args.urls` is non-empty then there are errors?
No. That won't be always the case. non-empty value for `args.urls` means a user has requested to send support data for specific URLs.